src/ott/solvers/nn/losses.py

```diff
@@ -19,13 +19,15 @@
 
 from ott.geometry import costs, pointcloud
 from ott.solvers.linear import sinkhorn
+from ott.solvers.nn import models
 
 __all__ = ["monge_gap"]
 
 
 def monge_gap(
-    source: jnp.ndarray,
-    target: jnp.ndarray,
+    model: models.ModelBase,
```

Comment:
I think this is too restrictive, it should be just a callable as before, removing the need for `params`.